### PR TITLE
Update infrastructure 20200126

### PR DESCRIPTION
**Pull request by phil-davis**
1) require PHP 7.1 or later and phpunit 7
2) test on PHP 7.1 through 7.4
3) switch to "standard" php-cs-fixer for code style
4) add `phpstan.neon` and run on CI at level 1 (which passes)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -43,8 +43,8 @@
         "psr/simple-cache-implementation": "~1.0"
     },
     "require-dev": {
-        "sabre/cs": "~1.0.0",
-        "phpunit/phpunit" : ">=6.2.4"
+        "friendsofphp/php-cs-fixer": "^2",
```
Comment (staabm) [inline composer.json]: I have bad experiences with this tool regarding BC - therefore I would recommend a non semver version constraint for it, like 2.16.* or similar

Produced codestyles highly depend on the minor version

**Comment by staabm** [comment]
Phpstan.neon is missing?

File: composer.json (hunk 1 above)
Comment (phil-davis) [reply composer.json]: Yes, there are sometimes new code style checks/fixers that are release as a minor version, but actually they find new stuff in existing code and so break CI.
Changed it.

**Comment by phil-davis** [comment]
> Phpstan.neon is missing?

I can see it in this PR.

**Comment by phil-davis** [comment]
@staabm ready for review again.

**Comment by phil-davis** [comment]
On PHP 7.3 it failed:
```
$ ./bin/phpunit --verbose --configuration tests/phpunit.xml.dist

PHPUnit 7.5.20 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.3.14 with Xdebug 2.9.1

Configuration: /home/travis/build/sabre-io/cache/tests/phpunit.xml.dist

.............................FF.........S........................ 65 / 72 ( 90%)

.......                                                           72 / 72 (100%)

Time: 12.07 seconds, Memory: 4.00 MB

There were 2 failures:

1) Sabre\Cache\MemcachedTest::testSetExpire

Failed asserting that null matches expected 'bar'.

/home/travis/build/sabre-io/cache/tests/AbstractCacheTest.php:80

2) Sabre\Cache\MemcachedTest::testSetExpireDateInterval

Failed asserting that null matches expected 'bar'.

/home/travis/build/sabre-io/cache/tests/AbstractCacheTest.php:96
```
Hmmm - why did it pass on 7.1 7.3 7.4 ???

**Comment by phil-davis** [comment]
@staabm no merge button for me here. Please merge.

**Pull request merged**